integration tests in `track/2.1` deploy dependency charms from `latest/edge`

### Bug Description

In branch `track/2.1`, integration test `test_deploy_resource_dispatcher` deploys `resource-dispatcher` and `metacontroller-operator` charms from channel `latest/edge`. The tests should rather pin the channels for these charms to their corresponding `<version>/stable` channels.

As a result, the integration tests are failing with the `juju.errors.JujuAPIError: no relations found` due to changes done in main to the relations interface in https://github.com/canonical/resource-dispatcher/pull/42

### To Reproduce

1. `git clone https://github.com/canonical/mlflow-operator.git`
2. `git checkout track/2.1`
3. `tox -vve integration -- --model testing`

### Environment

microk8s 1.25-strict/stable
juju 3.1/stable

### Relevant Log Output

```shell
=================================== FAILURES ===================================
__________________ TestCharm.test_deploy_resource_dispatcher ___________________
Traceback (most recent call last):
  File "/home/runner/work/mlflow-operator/mlflow-operator/.tox/integration/lib/python3.8/site-packages/_pytest/runner.py", line 341, in from_call
    result: Optional[TResult] = func()
  File "/home/runner/work/mlflow-operator/mlflow-operator/.tox/integration/lib/python3.8/site-packages/_pytest/runner.py", line 262, in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
  File "/home/runner/work/mlflow-operator/mlflow-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_hooks.py", line 493, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
  File "/home/runner/work/mlflow-operator/mlflow-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_manager.py", line 115, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/home/runner/work/mlflow-operator/mlflow-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_callers.py", line 152, in _multicall
    return outcome.get_result()
  File "/home/runner/work/mlflow-operator/mlflow-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_result.py", line 114, in get_result
    raise exc.with_traceback(exc.__traceback__)
  File "/home/runner/work/mlflow-operator/mlflow-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_callers.py", line 77, in _multicall
    res = hook_impl.function(*args)
  File "/home/runner/work/mlflow-operator/mlflow-operator/.tox/integration/lib/python3.8/site-packages/_pytest/runner.py", line 177, in pytest_runtest_call
    raise e
  File "/home/runner/work/mlflow-operator/mlflow-operator/.tox/integration/lib/python3.8/site-packages/_pytest/runner.py", line 169, in pytest_runtest_call
    item.runtest()
  File "/home/runner/work/mlflow-operator/mlflow-operator/.tox/integration/lib/python3.8/site-packages/_pytest/python.py", line 1792, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
  File "/home/runner/work/mlflow-operator/mlflow-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_hooks.py", line 493, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
  File "/home/runner/work/mlflow-operator/mlflow-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_manager.py", line 115, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/home/runner/work/mlflow-operator/mlflow-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_callers.py", line 152, in _multicall
    return outcome.get_result()
  File "/home/runner/work/mlflow-operator/mlflow-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_result.py", line 114, in get_result
    raise exc.with_traceback(exc.__traceback__)
  File "/home/runner/work/mlflow-operator/mlflow-operator/.tox/integration/lib/python3.8/site-packages/pluggy/_callers.py", line 77, in _multicall
    res = hook_impl.function(*args)
  File "/home/runner/work/mlflow-operator/mlflow-operator/.tox/integration/lib/python3.8/site-packages/_pytest/python.py", line 194, in pytest_pyfunc_call
    result = testfunction(**testargs)
  File "/home/runner/work/mlflow-operator/mlflow-operator/.tox/integration/lib/python3.8/site-packages/pytest_asyncio/plugin.py", line 532, in inner
    _loop.run_until_complete(task)
  File "/usr/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/home/runner/work/mlflow-operator/mlflow-operator/tests/integration/test_charm.py", line 324, in test_deploy_resource_dispatcher
    await ops_test.model.relate(
  File "/home/runner/work/mlflow-operator/mlflow-operator/.tox/integration/lib/python3.8/site-packages/juju/model.py", line 1559, in relate
    return await self.integrate(relation1, relation2)
  File "/home/runner/work/mlflow-operator/mlflow-operator/.tox/integration/lib/python3.8/site-packages/juju/model.py", line 1537, in integrate
    result = await app_facade.AddRelation(endpoints=endpoints, via_cidrs=None)
  File "/home/runner/work/mlflow-operator/mlflow-operator/.tox/integration/lib/python3.8/site-packages/juju/client/facade.py", line 486, in wrapper
    reply = await f(*args, **kwargs)
  File "/home/runner/work/mlflow-operator/mlflow-operator/.tox/integration/lib/python3.8/site-packages/juju/client/_client17.py", line 890, in AddRelation
    reply = await self.rpc(msg)
  File "/home/runner/work/mlflow-operator/mlflow-operator/.tox/integration/lib/python3.8/site-packages/juju/client/facade.py", line 659, in rpc
    result = await self.connection.rpc(msg, encoder=TypeEncoder)
  File "/home/runner/work/mlflow-operator/mlflow-operator/.tox/integration/lib/python3.8/site-packages/juju/client/connection.py", line 675, in rpc
    raise errors.JujuAPIError(result)
juju.errors.JujuAPIError: no relations found
```


### Additional Context

As seen in the [CI run](https://github.com/canonical/mlflow-operator/actions/runs/8174033900/job/22347864919) for #231

--- a/tests/integration/test_charm.py
+++ b/tests/integration/test_charm.py
@@ -299,7 +299,7 @@ class TestCharm:
     async def test_deploy_resource_dispatcher(self, ops_test: OpsTest):
         await ops_test.model.deploy(
             entity_url=METACONTROLLER_CHARM_NAME,
-            channel="latest/edge",
+            channel="3.0/stable",
             trust=True,
         )
         await ops_test.model.wait_for_idle(
@@ -310,7 +310,7 @@ class TestCharm:
             timeout=120,
         )
         await ops_test.model.deploy(
-            RESOURCE_DISPATCHER_CHARM_NAME, channel="latest/edge", trust=True
+            RESOURCE_DISPATCHER_CHARM_NAME, channel="1.0/stable", trust=True
         )
         await ops_test.model.wait_for_idle(
             apps=[CHARM_NAME],

